pop.go

```diff
@@ -56,7 +56,7 @@ func DialectSupported(d string) bool {
 	return false
 }
 
-func normalizeSynonyms(dialect string) string {
+func NormalizeSynonyms(dialect string) string {
```

Comment:
even though if we expose this as a public API, the name could be ambiguous

Reply:
That makes sense, I’ll rename it. This function is the only one allowing us to normalize DB types registered in pop. We need this when building custom migration boxes, for example

Reply:
DB types registered in pop? I think all dialects are registered at `AvailableDialects` (as its official name) and you can look up that to check when checking which dialect is supported, (I think you already know that) as the same as `buffalo new` shows.

And this function is just for finding aliased names for a shortcut to the official name such as `sqlite` to `sqlite3` so I am confusing then we need to use this... :thinking:  Since deprecating or changing exported function could be harder, I would like to keep it as private.

Reply:
https://github.com/gobuffalo/pop/blob/0fdefb2b4fa2c1efbdae2fa2f8f8828739dec76c/pop.go#L38 is private, we have functions like https://github.com/gobuffalo/pop/blob/0fdefb2b4fa2c1efbdae2fa2f8f8828739dec76c/pop.go#L50 but they do not have the synonyms available. When creating migration files with a custom migrator, people use `postgresql`, `postgres`, or a number of other synonyms. We need a way to resolve those because the pop migrator ignores these dialects. Let's please not make this bigger than it is. We're trying to export an already existing functionality that exposes something which is otherwise inaccessible because it is private. I've renamed the method to be more clear. I've been helping maintain this library for the past 3 years. That doesn't mean that there are exceptions, but it means that the change is done in good spirit and a bit of leeway would be appreciated :)

Reply:
Hmm, I see. I am actually not sure what is the custom migrator you mentioned but if that kind of situation could happen, it could help that situation maybe. Also, I don't want to be tight for code review but just wanted to make it clear so we can make sure how the function could be used by users, also with a better name when especially the change is about API change.

(Also, I know you've contributed a lot on the pop, even though most of my time on pop was too old and was not overlapped with yours, so I know your insight for pop would be much better than mine :-)

Reply:
Hi @paganotoni,

What do you think? It's about exposing an existing function (a small API change). Actually, I am not 100% sure but this change could be ok... (but I would like to see the real-world example for this though)

Reply:
Real-world example is here: 

https://github.com/ory/x/blob/b5bc5fda4c982d158b65e93bde7bf455dfad7157/popx/migration_box.go#L80

We use this migrator and pop across all of [Ory's stack](http://github.com/ory) (one of the biggest pop adopters). Due to some of our naming conventions we can't use `pop.ParseMigrationFiles` which uses this method internally

https://github.com/gobuffalo/pop/blob/5a8f51c37336105490d5c27c25ddf255886ec5d8/match.go#L17-L48

So it's basically impossible to write your own `ParseMigrationFilename` without copy-pasting this function (or exporting it).

As `DialectSupported` is exported, there's no reason not to export `NormalizeDialectSynonyms`.